modules/apps/jaggery/emm/src/emm/units/user-listing/public/js/user-listing.js

```diff
@@ -141,6 +141,74 @@ function removeUser(uname, uid) {
 }
 
 /**
+ * Following click function would execute
+ * when a user clicks on "Reset Password" link
+ * on User Listing page in WSO2 MDM Console.
+ */
+function resetPassword(uname) {
+
+    $(modalPopupContent).html($('#reset-password-window').html());
+    showPopup();
+
+    $("a#reset-password-yes-link").click(function () {
+        var newPassword = $("#new-password").val();
+        var confirmedPassword = $("#confirmed-password").val();
+        var user = uname;
+
+        var errorMsgWrapper = "#notification-error-msg";
+        var errorMsg = "#notification-error-msg span";
+        if (!newPassword) {
+            $(errorMsg).text("New password is a required field. It cannot be empty.");
+            $(errorMsgWrapper).removeClass("hidden");
+        } else if (!confirmedPassword) {
+            $(errorMsg).text("Retyping the new password is required.");
+            $(errorMsgWrapper).removeClass("hidden");
+        } else if (confirmedPassword != newPassword) {
+            $(errorMsg).text("New password doesn't match the confirmation.");
+            $(errorMsgWrapper).removeClass("hidden");
+        } else if (!inputIsValid(/^[\S]{5,30}$/, confirmedPassword)) {
+            $(errorMsg).text("Password should be minimum 5 characters long, should not include any whitespaces.");
+            $(errorMsgWrapper).removeClass("hidden");
+        } else {
+            var resetPasswordFormData = {};
+            resetPasswordFormData.username = user;
+            resetPasswordFormData.newPassword = window.btoa(unescape(encodeURIComponent(confirmedPassword)));
+
+            var resetPasswordAPI = "/mdm-admin/users/reset-password";
+
+            invokerUtil.post(
+                resetPasswordAPI,
```

Comment:
This doesn't sound like a good variable name. Something appropriate would be "resetPasswordServiceEpr" or something similar.

Reply:
Fixed